M365/src/DeleteContactsFromFolder/DeleteContactsFromFolder.ps1

```diff
@@ -0,0 +1,188 @@
+        04/27/2023  - Initial Public Release on CSS-Exchange
+#>
+
+$BuildVersion = ""
+
+. $PSScriptRoot\..\..\..\Shared\OutputOverrides\Write-Host.ps1
+. $PSScriptRoot\..\..\..\Shared\OutputOverrides\Write-Verbose.ps1
+. $PSScriptRoot\..\..\..\Shared\OutputOverrides\Write-Warning.ps1
+. $PSScriptRoot\..\..\..\Shared\ScriptUpdateFunctions\Test-ScriptVersion.ps1
+. $PSScriptRoot\..\..\..\Shared\LoggerFunctions.ps1
+. $PSScriptRoot\..\..\..\Shared\AzureFunctions\Get-GraphAccessToken.ps1
+. $PSScriptRoot\..\..\..\Shared\AzureFunctions\Invoke-GraphApiRequest.ps1
+
+function Write-DebugLog($Message) {
+    $Script:Logger = $Script:Logger | Write-LoggerInstance $Message
+}
+
+function Main {
+    if (Test-ScriptVersion -AutoUpdate -Confirm:$false) {
+        Write-Host ("Script was updated. Please re-run the command") -ForegroundColor Yellow
+        return
+    }
+
+    Write-Host ("DeleteContactsFromFolders.ps1 script version $($BuildVersion)") -ForegroundColor Green
+
+    $graphApiUrl = "https://graph.microsoft.com"
+    $unableToDeleteAllContacts = $false
+    $contacts = New-Object 'System.Collections.Generic.List[object]'
+
+    $getAccessTokenParams = @{
+        AzureADEndpoint = "https://login.live.com"
+        ClientId        = "7e791a92-18c9-4811-b48e-a0200052991c"
+        Scope           = "contacts.readwrite user.read mail.read"
+    }
+    $token = Get-GraphAccessToken @getAccessTokenParams
+
+    # Graph API call to get the current logged in user
+    $loggedInUserParams = @{
+        Query       = "me"
+        AccessToken = $token.AccessToken
+        GraphApiUrl = $graphApiUrl
+    }
+    $loggedInUserResponse = Invoke-GraphApiRequest @loggedInUserParams
+
+    if ($loggedInUserResponse.Successful -eq $false) {
+        Write-Host "Unable to query the logged in user. Please try again." -ForegroundColor Red
+        exit
+    }
+    Write-Host "Logged in user is: $($loggedInUserResponse.Content.UserPrincipalName)"
+
+    try {
+        # Get the list of contact folders
+        $listContactFolders = @{
+            Query       = "me/contactFolders"
+            AccessToken = $token.AccessToken
+            GraphApiUrl = $graphApiUrl
+        }
+        $listFoldersResponse = Invoke-GraphApiRequest @listContactFolders
+        if ($listFoldersResponse.Successful -eq $false) {
+            Write-Host "Unable to query the contact folders. Please try again." -ForegroundColor Red
+            exit
+        }
+
+        $folders = $listFoldersResponse.Content.value
+
+        if ($folders.Count -gt 0) {
+            Write-Host "List of existing folders:"
+            Write-Host "------------------------------------"
+            foreach ($folder in $folders) {
+                Write-Host $folder.displayName
+            }
+            Write-Host "------------------------------------"
+
+            # Get the folder name to be deleted
+            $folderToBeDeleted = Read-Host -Prompt "Enter the folder name to be deleted"
```

Comment:
We have rules in place that prevent `Read-Host` from being used. We shouldn't use them unless for good reason.

Reply:
This also doesn't handle if there are more than 1 folder that needs to be deleted.

Reply:
As discussed, we keep `Read-Host` in this script at this place.